Sources/Bugsnag/BugsnagReporter.swift

```diff
@@ -0,0 +1,306 @@
+            version: "3",
+            url: "https://github.com/nodes-vapor/bugsnag.git"
+        )
+
+        let abort = error as? AbortError
+        let reason = abort?.reason ?? "Something went wrong"
+        let status = abort?.status ?? .internalServerError
+
+        let exception = BugsnagException(
+            errorClass: error.localizedDescription,
+            message: reason,
+            stacktrace: [.init(
+                file: callsite.file,
+                lineNumber: callsite.line,
+                columnNumber: 0,
+                method: callsite.function,
+                inProject: true,
+                code: []
+            )],
+            type: status.reasonPhrase
+        )
+
+        let breadcrumbs = try req.privateContainer.make(BreadcrumbContainer.self)
+            .breadcrumbs
+
+        let http = req.http
+
+        var body: String? = nil
+        if let data = http.body.data {
+            body = String(data: data, encoding: .utf8)
+        }
+
+        let eventRequest = BugsnagRequest(
+            clientIp: http.remotePeer.hostname ?? "",
+            headers: parseHeaders(headers: http.headers),
+            httpMethod: "\(http.method)",
+            url: http.urlString,
+            referer: http.remotePeer.description,
+            body: body
+        )
+
+        var metadata: [String: String] = [
+            "Error localized description": error.localizedDescription,
+            "Request debug description": http.debugDescription
+        ]
+
+        metadata.reserveCapacity(2 + userMetadata.count)
+        for (key, value) in userMetadata {
+            metadata[key] = value.debugDescription
+        }
+
+        var user: BugsnagUser? = nil
+        if let id = userId {
+            user = BugsnagUser(id: "\(id)")
+        }
+
+        let event = BugsnagEvent(
+            payloadVersion: "4",
+            exceptions: [exception],
+            breadcrumbs: breadcrumbs,
+            request: eventRequest,
+            unhandled: true,
+            severity: severity,
+            user: user,
+            app: app,
+            metaData: BugsnagMetaData(meta: metadata)
+        )
+
+        let payload = BugsnagPayload(
+            apiKey: apiKey,
+            notifier: notifier,
+            events: [event]
+        )
+
+        return try JSONEncoder().encode(payload)
+    }
+
+    func report(
+        severity: String,
+        _ error: Error,
+        userId: Int?,
+        metadata: [String: CustomDebugStringConvertible],
+        callsite: (String, String, Int),
+        on request: Request
+    ) -> Future<Void> {
+        do {
+            let body = try buildBody(
+                request,
+                error: error,
+                severity: severity,
+                userId: userId,
+                userMetadata: metadata,
+                callsite: callsite
+            )
+
+            return HTTPClient.connect(hostname: "notify.bugsnag.com", on: request)
```

Comment:
`hostName` has been defined at the top (needs to be updated slightly to exclude scheme)